infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -1008,11 +1046,24 @@ public LogHeader getLogHeader() {
         private final FileChannel pendingTrimChannel;
         @NonNull
         private String fileName;
+
         private Map<Long, AddressMetaData> knownAddresses = new ConcurrentHashMap();
         private Set<Long> trimmedAddresses = Collections.newSetFromMap(new ConcurrentHashMap<>());
         private Set<Long> pendingTrims = Collections.newSetFromMap(new ConcurrentHashMap<>());
+        private volatile int refCount = 0;
 
 
+        public synchronized void retain() {
+            refCount++;
+        }
+
+        public synchronized void release() {
```

Comment:
I've a question about the refcounting and `StreamLogFiles::close`, which isn't paying attention to the refcounts.  AFAICT, that method is only being used by unit test code, and honoring the refcounts in those tests probably doesn't matter?